### PR TITLE
Add forms

### PR DESCRIPTION
**Pull request by jonas-45**
In this milestone, Forms were added to enable users to add posts and comments
A like post button is added to allow users to like posts
To achieve the above, a number of controllers and the views for the corresponding actions were also created. The actions that were considered in this milestone are;

- [ ] new
- [ ] create

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -1,5 +1,7 @@
 class ApplicationController < ActionController::Base
   def current_user
-    User.first
+    @current_user ||= User.first
   end
+
+  helper_method :curren_user
```
Comment (Deepakdanger) [inline app/controllers/application_controller.rb]: - [ ] Here is a typo error kindly use `current_user`. Is this really required? Have you used `current_user` in views somewhere?

File: app/controllers/application_controller.rb [hunk 2]
```diff
@@ -1,5 +1,7 @@
 class ApplicationController < ActionController::Base
   def current_user
-    User.first
+    @current_user ||= User.first
```
Comment (Deepakdanger) [inline app/controllers/application_controller.rb]: - [ ] You have not used `@current_user` anywhere, so kindly replace it with `User.first` as everywhere its calling `current_user` method only.

File: app/controllers/posts_controller.rb [hunk 3]
```diff
@@ -9,4 +9,29 @@ def show
...
+
+  def new
+    @post = Post.new
+  end
+
+  def create
+    @post = Post.new(post_params)
+    @post.author = current_user
+    @post.comments_counter = 0
+    @post.likes_counter = 0
```
Comment (Deepakdanger) [inline app/controllers/posts_controller.rb]: - [ ]  [OPTIONAL] In db level we have have set `comments_counter & likes_counter` as default 0, so we don't have to set here.

**Review by Deepakdanger — CHANGES_REQUESTED** [review]
Hi @jonas-45 , 

Good job so far! 👏
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight :100: 
- No linter error :heavy_check_mark: 
- Website looks professional :+1: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/application_controller.rb (hunk 1 above)
Comment (jonas-45) [reply app/controllers/application_controller.rb]: Thank, I will review it right away

**Comment by jonas-45** [comment]
Thanks @Deepakdanger for your effort and time in looking through my work. Will implement changes right away.

File: app/controllers/users_controller.rb [hunk 4]
```diff
@@ -5,7 +5,7 @@ def index
 
   def show
     @user = User.find(params[:id])
-    # @user_posts = Post.where(author_id: @user.id)
+    @post = Post.find(params[:id])
```
Comment (Kamba56) [inline app/controllers/users_controller.rb]: - [Optional] Kindly remove this line of code as it brings an error to your code. The users show action should only display details of the user and its three most recent posts. There is no post id in the parameters to show a single post so kindly remove this line of code.
image

**Review by Kamba56 — APPROVED** [review]
Hi @jonas-45, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏
## Highlights
- [x] Great job creating posts for the current user!👏👏👏
- [x] Great job working with forms in rails👏👏👏
- [x!👏👏👏

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/users_controller.rb (hunk 4 above)
Comment (jonas-45) [reply app/controllers/users_controller.rb]: Okay thanks

**Pull request merged**